### PR TITLE
Implement Unet3DLightning class

### PR DESCRIPTION
**Pull request by abhibyreddi**
Cloned `layers.py`, `losses.py`, `arguments.py`, and `unet3d.py` from the original demo code. 

Ran `python3  train.py --gcp_project=amundson-gke-aiml-demo --images_prefix=dataset-original/processed-data/images --labels_prefix=dataset-original/processed-data/labels --gcs_bucket=dlio-compute-time-bucket` locally and verified that the training ran for 2 full epochs without any errors.

### CODE REVIEW

File: demo/lightning/image-segmentation/train.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-    train_data_loader = Unet3DDataModule()
-    trainer = pl.Trainer(
-        devices=2, 
-        accelerator="gpu", 
-        max_epochs=5
-    )
+    flags = PARSER.parse_args()
+    model = Unet3DLightning(flags)
+    train_data_loader = Unet3DDataModule(flags)
+    trainer = pl.Trainer(accelerator="cpu", max_epochs=5)
```
Comment (MattIrv) [inline demo/lightning/image-segmentation/train.py:11]: Can you make these arguments configurable?

File: demo/lightning/image-segmentation/train.py [hunk 2]
```diff
@@ -1,14 +1,12 @@
 import lightning as pl
 
-from model import Unet3D
+from model import Unet3DLightning
 from data import Unet3DDataModule
+from arguments import PARSER
 
 if __name__ == "__main__":
```
Comment (MattIrv) [inline demo/lightning/image-segmentation/train.py:7]: Can you add a readme file to this demo explaining how to set up and run the demo and any additional information about it? It can come in a separate PR if that's easier.

**Review by MattIrv — APPROVED** [review]

File: demo/lightning/image-segmentation/train.py (hunk 2 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/train.py:7]: I will send out a separate PR that adds the instructions.

File: demo/lightning/image-segmentation/train.py (hunk 1 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/train.py]: Done! Added a new argument `accelerator` which can be set to the accelerator type. It defaults to `cpu`.
